src/util/sandbox.cpp

```diff
@@ -63,29 +63,31 @@ void Sandbox::shutdown() {
 }
 
 // static
-bool Sandbox::askForAccess(const QString& canonicalPath) {
+bool Sandbox::askForAccess(const QString& path) {
```

Comment:
I don't think that we should drop the assumption that this is indeed a canonical path.

Instead we should add a debug assertion that this precondition and assumption is true:
```
DEBUG_ASSERT(QFileInfo(canonicalPath).canonicalFilePath() == canonicalPath);
```

Reply:
The preferred solution would be to do the normalization in here and use an `enum class` for returning the result that should include `Nonexistent`/`NotFound`.

Reply:
This don't need to be canonical, and it should not, for better user interaction.  
The enum solution is a good idea, but to risky for 2.3 and I have no Mac for testing.

Reply:
Ok, then let's assume this was semantically overqualified.